src/runtime.rs

```diff
@@ -130,7 +131,17 @@ impl FutureIntoCoroutine {
             // Here we have left the trio thread and are inside a thread provided by the Tokio runtime
 
             // Actual run of the Tokio future
-            let ret = fut.await;
+            let ret = std::panic::AssertUnwindSafe(fut).catch_unwind().await;
+            let ret = ret.unwrap_or_else(|panic_err| {
+                let msg = match panic_err.downcast::<&str>() {
+                    Ok(msg) => (*msg).to_owned(),
+                    Err(panic_err) => match panic_err.downcast::<String>() {
+                        Ok(msg) => *msg,
+                        Err(_) => "Unknown error".to_owned(),
```

Comment:
I don't think we would fall on this step often, but providing additional information would be useful if it's the case :thinking: 

```suggestion
                        Err(_) => format!("Unknown error type {:?}", panic_err.type_id())
```

Reply:
That's a very good idea !
Having a weak clue is much much better than not having any clue here ^^